docs/component_conventions.md

```diff
@@ -0,0 +1,68 @@
+# Component Conventions
+
+All components must use the `.app-` or `.pub-` namespace with the `c` prefix. The `c` is for component. For example, `.app-c-banner`.
+
+Do not use the `.govuk-c` namespace.
+
+The namespace indicates where a component lives. A single page on GOV.UK could render components from multiple places.
+
+| Prefix | Place |
+| -- | -- |
+| `.app-c-` | Component lives within the frontend application |
+| `.pub-c-` | Component is shared between publishing frontends and lives in static |
+| `.govuk-c-` | Component originates from [GOV.UK Frontend](https://github.com/alphagov/govuk-frontend) |
```

Comment:
I think the confusing thing is that styles are discussed twice. I think all the style stuff should be together.